File: Medium/2025-11-03-1578-Minimum-Time-to-Make-Rope-Colorful/solution.js
var minCost = function(colors, neededTime) {
    let totalCost = 0;
    let maxTime = 0;
    
    for (let i = 0; i < colors.length; i++) {
        if (i > 0 && colors[i] === colors[i - 1]) {
            totalCost += Math.min(maxTime, neededTime[i]);
            maxTime = Math.max(maxTime, neededTime[i]);
        } else {
            maxTime = neededTime[i];
        }
    }
    
    return totalCost;
};

File: Medium/2025-11-03-1578-Minimum-Time-to-Make-Rope-Colorful/solution.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int totalCost = 0;
        int maxTime = 0;
        int n = colors.length();
        
        for (int i = 0; i < n; i++) {
            if (i > 0 && colors.charAt(i) == colors.charAt(i - 1)) {
                totalCost += Math.min(maxTime, neededTime[i]);
                maxTime = Math.max(maxTime, neededTime[i]);
            } else {
                maxTime = neededTime[i];
            }
        }
        
        return totalCost;
    }
}